app/src/main/java/com/akshay/newsapp/news/ui/activity/NewsActivity.kt

```diff
@@ -29,7 +30,11 @@ class NewsActivity : BaseActivity() {
         newsList.setEmptyView(empty_view)
         newsList.setProgressView(progress_view)
 
-        val adapter = NewsArticlesAdapter { toast("Clicked on item") }
+        val adapter = NewsArticlesAdapter {
+            val intent = Intent(this, NewsDetailActivity::class.java)
+            intent.putExtras(Bundle().apply { putParcelable("details", it) })
+            startActivity(intent)
+        }
```

Comment:
I'd recommend to not do this pattern, it definitely seems easy but it's not great especially for this use case. 
My suggestion is to 
1. Pass the "id" of the news article through the intent
2. Get the ID In`NewsDetailActivity`
3. Fetch the news article from repository using the ID

Reply:
Also would be nice if we could use `intentOf` extension function provided in [Android KTX](https://developer.android.com/kotlin/ktx)